### PR TITLE
EstimateLibraryComplexity should ignore secondary or supplementary

### PR DESCRIPTION
**Pull request by nh13**
records.

The documentation for `INPUT` states the data can be mapped or unmapped.  In the former case, we could have secondary or supplementary records.  This may lead to over-estimating the number of optical duplicates, as the same read is seen multiple times.  I added a test to cover ignoring secondary and supplementary records. 

@yfarjoun @tfenne

### CODE REVIEW

File: src/tests/java/picard/sam/markduplicates/EstimateLibraryComplexityTest.java [hunk 1]
```diff
@@ -67,6 +67,22 @@ public void testSimpleDuplicate() throws IOException {
...
     }
 
+    /** Finds duplicates as expected ignoring secondary and supplementary records. */
+    @Test
+    public void testSimpleDuplicatesWithSecondaryAndSupplementaryRecords() throws IOException {
+        final File input = new File(TEST_DATA_DIR, "dupes_with_sos.sam");
+        final File output = File.createTempFile("estimateLibraryComplexity",".els_metrics");
+        output.deleteOnExit();
+
+        final List<String> args =new ArrayList<String>();
```
Comment (yfarjoun) [inline src/tests/java/picard/sam/markduplicates/EstimateLibraryComplexityTest.java]: spaces (ha!)

Comment (yfarjoun) [inline src/tests/java/picard/sam/markduplicates/EstimateLibraryComplexityTest.java]: and java8

**Comment by yfarjoun** [comment]
99/💯 thanks. Space and java8 comments and this is fine by me.

File: src/tests/java/picard/sam/markduplicates/EstimateLibraryComplexityTest.java (hunk 1 above)
Comment (nh13) [inline src/tests/java/picard/sam/markduplicates/EstimateLibraryComplexityTest.java]: I'm melting.

**Comment by nh13** [comment]
@yfarjoun back to you.

File: src/tests/java/picard/sam/markduplicates/EstimateLibraryComplexityTest.java [hunk 2]
```diff
@@ -74,7 +90,7 @@ public void testMaxDiffRate() throws IOException {
         final File output = File.createTempFile("estimateLibraryComplexity",".els_metrics");
         output.deleteOnExit();
 
-        final List<String> args =new ArrayList<String>();
+        final List<String> args = new ArrayList<>();
```
Comment (nh13) [inline src/tests/java/picard/sam/markduplicates/EstimateLibraryComplexityTest.java]: there's just so much space

**Comment by yfarjoun** [comment]
could you also please modify the documentation to inform that these reads will be ignored ?

**Comment by nh13** [comment]
@yfarjoun done, thanks for the review, ready to merge after tests pass.

**Comment by yfarjoun** [comment]
👍

**Pull request merged**